llama_stack/providers/adapters/inference/vllm/vllm.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+from typing import AsyncGenerator
+
+from llama_models.llama3.api.chat_format import ChatFormat
+
+from llama_models.llama3.api.datatypes import Message
+from llama_models.llama3.api.tokenizer import Tokenizer
+
+from openai import OpenAI
+
+from llama_stack.apis.inference import *  # noqa: F403
+
+from llama_stack.providers.utils.inference.model_registry import ModelRegistryHelper
+from llama_stack.providers.utils.inference.openai_compat import (
+    get_sampling_options,
+    process_chat_completion_response,
+    process_chat_completion_stream_response,
+)
+from llama_stack.providers.utils.inference.prompt_adapter import (
+    chat_completion_request_to_prompt,
+)
+
+from .config import VLLMImplConfig
+
+
+# Reference: https://docs.vllm.ai/en/latest/models/supported_models.html
+VLLM_SUPPORTED_MODELS = {
+    "Llama3.1-8B-Instruct": "meta-llama/Meta-Llama-3.1-8B-Instruct",
```

Comment:
so I don't think we should assume a fixed list because this is a vLLM server someone is running on their own. can you query the vLLM server and find out the model it is serving? you should do this via the `list_models()` function (and we should not be using ModelRegistryHelper) -- see the `ollama` or `tgi` provider for how this is done.

Reply:
I switched to reuse `all_registered_models()` for this

Reply:
I ended up keeping this list because it's easier to map the models but implemented `list_models()` with OpenAI client.